redfish-core/lib/processor.hpp

```diff
@@ -975,6 +995,280 @@ inline void
+            }
+        }
+    }
+}
+
+inline void
+    getCpuCoreDataByService(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& service,
+                            const std::string& objPath)
+{
+    BMCWEB_LOG_DEBUG("Get available system cpu core resources by service.");
+
+    asyncResp->res.jsonValue["Status"]["State"] = "Enabled";
+    asyncResp->res.jsonValue["Status"]["Health"] = "OK";
+
+    dbus::utility::getManagedObjects(
+        service,
+        sdbusplus::message::object_path("/xyz/openbmc_project/inventory"),
+        std::bind_front(afterGetCpuCoreDataByService, asyncResp, objPath));
+}
+
+inline void aferGetSubProcessorData(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, const std::string& coreId,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreeResponse& subtree)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error, ec: {}", ec.value());
+        // No processor objects found by mapper
+        if (ec.value() == boost::system::errc::io_error)
+        {
+            messages::resourceNotFound(
+                asyncResp->res, "#Processor.v1_18_0.Processor", processorId);
+            return;
+        }
+
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    for (const auto& object : subtree)
+    {
+        if (sdbusplus::message::object_path(object.first).filename() != coreId)
+        {
+            continue;
+        }
+
+        asyncResp->res.jsonValue["@odata.type"] =
+            "#Processor.v1_18_0.Processor";
+        asyncResp->res.jsonValue["@odata.id"] = boost::urls::format(
+            "/redfish/v1/Systems/system/Processors/{}/SubProcessors/{}",
+            processorId, coreId);
+        asyncResp->res.jsonValue["Name"] = "SubProcessor";
+        asyncResp->res.jsonValue["Id"] = coreId;
+
+        for (const auto& service : object.second)
+        {
+            getCpuCoreDataByService(asyncResp, service.first, object.first);
+            break;
+        }
+        return;
+    }
+
+    // Object not found
+    messages::resourceNotFound(asyncResp->res, "#Processor.v1_18_0.Processor",
+                               coreId);
+}
+
+inline void
+    getSubProcessorData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& processorId,
+                        const std::string& coreId)
+{
+    auto callback = [asyncResp, processorId,
+                     coreId](const std::string& cpuPath) {
+        dbus::utility::getSubTree(cpuPath, 0, cpuCoreIfaces,
+                                  std::bind_front(aferGetSubProcessorData,
+                                                  asyncResp, processorId,
+                                                  coreId));
+    };
+
+    getProcessorPaths(asyncResp, processorId, std::move(callback));
```

Comment:
Since the callback is small, can it be done inside lambda like
```
inline void
    getSubProcessorData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                        const std::string& processorId,
                        const std::string& coreId)
{
    getProcessorPaths(asyncResp, processorId, 
        [asyncResp, processorId, coreId](const std::string& cpuPath) {
               dbus::utility::getSubTree(cpuPath, 0, cpuCoreIfaces,
                   std::bind_front(aferGetSubProcessorData,
                   asyncResp, processorId, coreId));
        });
```